File: myadmin-angular/projects/temp-study-material-app/src/app/services/study-material.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {ModuleEnrolmentInfo, StudyMaterialDetailInfo} from "../info-objects";
import {StudentInfo} from "myadmin-lib";

@Injectable()
export class StudyMaterialService {

  constructor(private http:HttpClient) {
  }

  requestStudentModuleEnrolments(studentInfo:StudentInfo):Observable<ModuleEnrolmentInfo[]> {
    return this.http.post<ModuleEnrolmentInfo[]>('/myadmin-student-services/services/rest/moduleenrolmentservice/moduleEnrolment/courselist', studentInfo);
  }

  requestModuleStudyMaterials(moduleCode:string, academicYear:number, semesterCode:string):Observable<StudyMaterialDetailInfo[]> {
    return this.http.get<StudyMaterialDetailInfo[]>('/myadmin-student-services/services/rest/studymaterialservice/studymaterial/viewmaterial', {
      params: {
        moduleCode: moduleCode,
        academicYear: <string><any>academicYear,
        semesterCode: semesterCode
      }
    });
  }
}

File: myadmin-service-api/src/main/java/za/ac/unisa/myadmin/studymaterial/services/StudyMaterialService.java
package za.ac.unisa.myadmin.studymaterial.services;

import za.ac.unisa.myadmin.common.exceptions.DoesNotExistException;
import za.ac.unisa.myadmin.common.exceptions.InvalidParameterException;
import za.ac.unisa.myadmin.common.exceptions.MissingParameterException;
import za.ac.unisa.myadmin.common.exceptions.OperationFailedException;
import za.ac.unisa.myadmin.studymaterial.services.dto.StudyMaterialDetailInfo;

import java.util.List;

public interface StudyMaterialService {

	public List<StudyMaterialDetailInfo> getModuleStudyMaterials(String moduleCode, Integer academicYear, String semesterCode) throws MissingParameterException, InvalidParameterException, OperationFailedException, DoesNotExistException;
}
